core/src/main/java/org/elasticsearch/index/query/DisMaxQueryBuilder.java

```diff
@@ -169,8 +168,8 @@ public static DisMaxQueryBuilder fromXContent(QueryParseContext parseContext) th
             }
         }
 
-        if (!queriesFound) {
-            throw new ParsingException(parser.getTokenLocation(), "[dis_max] requires 'queries' field");
+        if (!queriesFound || queries.isEmpty()) {
```

Comment:
good point, the old behaviour was to return no documents when there are no clauses (via returning null in toQuery for this case). I will change this back here and return MatchNoDocsQuery in toQuery() for this edge case.